src/routes/safe/store/actions/utils.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import axios from 'axios'
+
+import { buildTxServiceUrl } from '~/logic/safe/transactions/txHistory'
+import { getGnosisSafeInstanceAt } from '~/logic/contracts/safeContracts'
+
+export const getLastTx = async (safeAddress: string): Promise<TransactionProps> => {
+  try {
+    const url = buildTxServiceUrl(safeAddress)
+    const response = await axios.get(url, { params: { limit: 1 } })
+
+    return response.data.results[0]
+  } catch (e) {
+    console.error('failed to retrieve last Tx from server', e)
+    return null
+  }
+}
+
+export const getSafeNonce = async (safeAddress: string): Promise<string> => {
+  // use current's safe nonce as fallback
+  const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+  return (await safeInstance.nonce()).toString()
+}
+
+export const getNewTxNonce = async (txNonce, lastTx, safeAddress) => {
```

Comment:
can you also pass safeInstance there instead of safeAddress? you get the instance there before calling this function https://github.com/gnosis/safe-react/blob/2e66e4b502859739dad8107d5fb15e3d087c2758/src/routes/safe/store/actions/createTransaction.js#L58

Reply:
this also may be the case for processTransaction.js

Reply:
:100: and that was a really good catch. Thanks!